### PR TITLE
Breadcrumbs: Separate mobile breadcrumbs from tabs

### LINKED ISSUE #415: Hidden mobile breadcrumbs aren't useful to users
image

## History
The breadcrumbs were abstracted behind the button with ellipses ⬆️ because they would get cut off on mobile and look bad. https://github.com/iFixit/react-commerce/issues/201

### TLDR 
The breadcrumb button is not obvious, and users got frustrated trying to figure out how to navigate back up the hierarchy. It was easy to navigate down and use the search bar from high level categories like `/Parts` or `/Tools`, but once they got too deep and needed to back out, they couldn't get there.

### Details
There was a clear problem noted by every tester: It's difficult to navigate back up the device hierarchy. It was almost impossible on mobile. 

Almost every single person who tested on mobile didn't realize that breadcrumbs were available to them. 

Several users stated that they would expect some hierarchy here, but didn't know how to use it.

A few noted that the ellipses button looked like an admin setting button or generally not something meant for them. 

Users got frustrated. Some noted that this might be the point at which they give up on trying to find a product. One asked "Am I just a dumb user?"

Other ways users tried to navigate back up the hierarchy when they got blocked:
- Navigating to a product page where they know there are breadcrumbs and then using those to get to the live /Store/Parts page
- Clicking on the "Parts" slider of Parts/Guides/Answers
- Manipulating the filters (One user, who entered on a nested category page, suspected that there might be some filters that were set without them knowing, which caused them to have limited results on their current page)

### LINKED ISSUE #470: Empty toolbar on mobile /Tools
We removed the Parts/Guides/Answers slider on `/Tools`. On `/Tools` children pages, that header contains the breadcrumbs, but it's empty and looks odd on the root page - `/Tools`

| Root page (/Tools) | Child collection page |
| --- | --- |
| image | image |

### PR DESCRIPTION
**Pull request by masonmcelvain**
Puts the breadcrumbs on a new navbar below the Parts/Guides/Answers tab
on mobile browsers. Always shows the `> Last Breadcrumb` text, because
the ellipsis by themselves were deemed confusing by users.

Design:
https://www.figma.com/file/sfj6r7Bn6eREMvuqo0f7Du/iFixit---Components?node-id=206%3A15699
https://www.figma.com/file/sfj6r7Bn6eREMvuqo0f7Du/iFixit---Components?node-id=118%3A12832

## QA

Make sure the preview looks like the Figma, and that the UX is as expected and the links all work (at least compared to main, the All Parts still links to /Store/Parts).

CC @erinemay @mmarcon91 @sterlinghirsh 

Closes #415
Closes #470

### CODE REVIEW

File: frontend/components/product-list/ProductListBreadcrumb.tsx [hunk 1]
```diff
@@ -40,10 +40,7 @@ export function ProductListBreadcrumb({
          separator={
             <Icon
                as={HiChevronRight}
-               display={{
-                  base: 'none',
-                  md: 'initial',
-               }}
+               display="initial"
```
Comment (sterlinghirsh) [inline frontend/components/product-list/ProductListBreadcrumb.tsx:43]: Do we need this anymore if we're never setting it to none?

**Review by sterlinghirsh — DISMISSED** [review]

**Comment by sterlinghirsh** [comment]
CR :zap:

File: frontend/components/product-list/ProductListBreadcrumb.tsx (hunk 1 above)
Comment (masonmcelvain) [reply frontend/components/product-list/ProductListBreadcrumb.tsx:43]: Nope good catch

Comment (masonmcelvain) [reply frontend/components/product-list/ProductListBreadcrumb.tsx]: https://github.com/iFixit/react-commerce/pull/466/commits/ebb2f7d82e232fad6c85f2e63aec017ce62fa07a

**Comment by erinemay** [comment]
dev_block 🧙 on a couple of things

<details><summary>On desktop, there's an initial flash of the second toolbar, the footer, and the open facets before the page appears as expected.</summary>

https://user-images.githubusercontent.com/72166715/180266178-565f472b-d693-4b0e-8024-079aab85876d.mp4

</details>

<details><summary>Nitpick: Can we move the no-breadcrumbs text over a couple of pixels to be in line with the slider under "Parts"? (Also, see deploy_block comment about no-breadcrumb text ⬇️)</summary>

It's like this across different mobile devices.

| No breadcrumbs | Breadcrumbs |
| -- | -- |
| image | image |

</details>

<details><summary>What's the logic behind only showing only 3 breadcrumbs?</summary>
I think there should always be a direct link back to the root category available. Without it, you have to play a jumping game of getting 3 higher until the "All Parts" breadcrumb shows up.

| Deep nested category | <- its parent | <- its parent |
| --- | --- | --- |
| image | image | image |

</details>

<details><summary>Some collection titles are getting cut off</summary>
This is fairly edge. Most collection titles fit just fine within the space allotted. It's the longer ones with model numbers like this one that get cut off. Should we add ellipses at the end? 

image

</details>

And deploy_block ❔ on a question and one thing that might be out of scope for this pull: 

<details><summary>Should we show the bar on /Parts or /Tools?</summary>

- Keeping it: It feels structurally consistent. It would be less pleasing to open a collection and have the structure of the page change.
- Dropping it: It looks a little silly and redundant without any breadcrumbs.

I don't feel super strongly either way, just noting the sort of odd look of it without any breadcrumbs. It may still be the best option.

image

</details>

<details><summary>Empty toolbar on /Tools (mayyyybe out of scope)</summary>
I wrote up an issue for the empty toolbar on /Tools: /issues/470

It's technically it's own issue, but the effect of the changes here on /Tools children collection pages makes me more inclined to include it here.

| This branch | Production |
| --- | --- |
| image | image |

</details>

Otherwise, confirmed:
- The breadcrumb button is clickable and displays breadcrumbs that are also clickable ✔️ 
- Accessibility: Works with a screen reader and gestures on Android and iOS ✔️

**Comment by masonmcelvain** [comment]
Thanks @erinemay!

> Empty toolbar on /Tools (mayyyybe out of scope)

Now that we're hiding the nav bar on /Tools, I think that fixes the problem? Let me know if I misunderstand what the UI issue is.

I think the rest of the problems should be addressed. un_dev_block ✌🏻 un_deploy_block ✌🏻

**Comment by sterlinghirsh** [comment]
CR :zap:

**Review by sterlinghirsh — APPROVED** [review]

**Comment by erinemay** [comment]
> Now that we're hiding the nav bar on /Tools, I think that fixes the problem? Let me know if I misunderstand what the UI issue is.

I'm not sure...

Are we planning on dropping the entire nav bar on /Tools? Should there be an issue for that? Is there a plan to allow navigation back to /Tools from sub-categories? 

If that's not going to happen before release, having an extra, empty nav bar should be fixed. I made https://github.com/iFixit/react-commerce/issues/470 to address the issue of an empty nav bar on the root /Tools page, but the doubling up does appear to be introduced here. I'm still seeing it in the latest deploy. This ⬇️ is on /Tools/iFixit_Gear:

| This branch | Main/Production |
| --- | --- |
| image | image |

**Comment by masonmcelvain** [comment]
You're right, the empty nav bar is weird. Is this more like what you would expect? In https://github.com/iFixit/react-commerce/issues/376 we hid navigation on all /Tools/** pages, so I think keeping it that way is what we want.


| /Tools | /Tools/iFixit_Gear |
| --- | --- |
| image | image |

**Comment by erinemay** [comment]
> You're right, the empty nav bar is weird. Is this more like what you would expect? In #376 we hid navigation (ie Parts/Guides/Answers) on all /Tools/** pages, so I think keeping it that way is what we want.

Yeah, that's more what I imagined.

**Comment by sterlinghirsh** [comment]
CR :zap:

**Review by sterlinghirsh — APPROVED** [review]

**Comment by jordycosta** [comment]
QA 🟢 

UX is as expected and the links all work 👍 
> Ellipses for longer titles are present
> No extra bar on /Tools
> No bar on root pages
> Up to 6 page history

---

### Testing Procedure:
1. Checkout | merge with [`breadcrumbs--redesign-for-mobile`](https://github.com/iFixit/react-commerce/tree/breadcrumbs--redesign-for-mobile)
2. Navigate to [react-commerce-prod](https://react-commerce-prod-git-breadcrumbs-redesign-for-mobile-ifixit.vercel.app/Parts)
3. Switch to mobile view
4. Ensure that navigation breadcrumbs look and function well
5. Verify the issues shown in [#issuecomment-1191744032](https://github.com/iFixit/react-commerce/pull/466#issuecomment-1192702268) are resolved

---

### Screen Captures:

breadcrumbs

Parts

Tools

**Pull request merged**